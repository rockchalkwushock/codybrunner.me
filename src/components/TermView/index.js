import React from 'react'

import Card from '../Card'
import { FlexContainer, Grid, List, Link } from '../commons'

// TODO: Need to create ternary in the event no posts exist for term, etc.

const Term = ({ tag, taggedPosts }) =>
  taggedPosts ? (
    <FlexContainer>
      <Grid alternate>
        <h1>
          {taggedPosts.length} post{taggedPosts.length === 1 ? '' : 's'} tagged
          with {tag}
        </h1>
        <List>
          {taggedPosts.map(post => <Card key={post.fields.slug} post={post} />)}
        </List>
      </Grid>
    </FlexContainer>
  ) : (
    <FlexContainer>
      <Grid>
        <p>Whoa warrior, woah! How did you do that? Maybe go home okay.</p>
        <Link href="/" text="Go back home" />
      </Grid>
    </FlexContainer>
  )

export default Term
