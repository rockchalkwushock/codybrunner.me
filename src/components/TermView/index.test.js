import React from 'react'
import { shallow } from 'enzyme'

import TermView from './'

const taggedPosts = [
  {
    excerpt: 'blah blah',
    fields: {
      slug: '/2018-02-03/all-about-js'
    },
    frontmatter: {
      date: 'February 03, 2018',
      draft: false,
      tags: ['JavaScript', 'fp', 'oop'],
      title: 'All About JS'
    }
  }
]

describe('Component: <TermView />', () => {
  let withTaggedPosts
  let noTaggedPosts
  beforeEach(() => {
    withTaggedPosts = shallow(
      <TermView tag="JavaScript" taggedPosts={taggedPosts} />
    )
    noTaggedPosts = shallow(<TermView tag="JavaScript" taggedPosts={null} />)
  })

  test('should render correctly with taggedPosts', () => {
    expect(withTaggedPosts).toMatchSnapshot()
  })
  test('should render correctly with no taggedPosts', () => {
    expect(noTaggedPosts).toMatchSnapshot()
  })
  test('should render tagged posts if posts are present', () => {
    expect(withTaggedPosts.find('List').children()).toHaveLength(1)
  })
  test('should render link back home if no posts found for term', () => {
    expect(noTaggedPosts.find('SiteLink')).toBeDefined()
    expect(noTaggedPosts.find('SiteLink').prop('text')).toEqual('Go back home')
  })
})
