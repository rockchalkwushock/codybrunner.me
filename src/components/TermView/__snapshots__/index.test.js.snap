// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TermView /> should render correctly with no taggedPosts 1`] = `
<FlexContainer>
  <Grid>
    <p>
      Whoa warrior, woah! How did you do that? Maybe go home okay.
    </p>
    <SiteLink
      href="/"
      text="Go back home"
    />
  </Grid>
</FlexContainer>
`;

exports[`Component: <TermView /> should render correctly with taggedPosts 1`] = `
<FlexContainer>
  <Grid
    alternate={true}
  >
    <h1>
      1
       post
       tagged with 
      JavaScript
    </h1>
    <List
      flow="row"
      gap={20}
    >
      <SiteCard
        key="/2018-02-03/all-about-js"
        post={
          Object {
            "excerpt": "blah blah",
            "fields": Object {
              "slug": "/2018-02-03/all-about-js",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
    </List>
  </Grid>
</FlexContainer>
`;
